models/Asset/Image/Thumbnail.php

```diff
@@ -396,6 +396,12 @@ public function getMedia($name, $highRes = 1)
      */
     protected function createConfig($selector)
     {
-        return Thumbnail\Config::getByAutoDetect($selector);
+        $thumbnailConfig = Thumbnail\Config::getByAutoDetect($selector);
+
+        if(!empty($selector) && $thumbnailConfig === null) {
+            throw new \Exception('Thumbnail definition "' . (is_string($selector)? $selector:'') . '" does not exist');
```

Comment:
I would prefer `Pimcore\Model\Exception\NotFoundException` and add it to the PHPDoc (also the other exceptions)

Reply:
Agree.

Reply:
Fixed. Thanks @blankse